crates/typst-library/src/visualize/polygon.rs

```diff
@@ -91,3 +97,55 @@ impl Layout for PolygonElem {
         Ok(Fragment::frame(frame))
     }
 }
+
+impl Reflect for PolygonElem {
+    fn describe() -> CastInfo {
+        CastInfo::Any
+    }
+
+    fn castable(_: &Value) -> bool {
+        false
+    }
+}
+
+/// A regular polygon (https://en.wikipedia.org/wiki/Regular_polygon).
+///
+/// The regular polygon is defined by its size and quantity of vertices.
+///
+/// ## Example { #example }
+/// ```example
+/// #polygon.regular(
+///   fill: blue.lighten(80%),
+///   stroke: blue,
+///   size: 30pt,
+///   vertices: 3,
+/// )
+/// ```
+///
+/// Display: Regular Polygon
+/// Category: visualize
+#[func]
+pub fn regular_polygon(
+    #[named] fill: Option<Paint>,
+    #[default(Smart::Auto)]
+    #[named]
+    stroke: Smart<Option<PartialStroke>>,
+    #[default(Length::from(Em::new(1.0)))]
+    #[named]
+    size: Length,
```

Comment:
This is feeling a bit cluttered imo
```suggestion
    stroke: Smart<Option<PartialStroke>>,

    #[default(Length::from(Em::new(1.0)))]
    #[named]
    size: Length,

```